### PR TITLE
chore: update SigLIP2 model card

### PR DESCRIPTION
**Pull request by saswatmeher**
# What does this PR do?
Update the model card for SigLIP2 to handle #36979 

Fixes #36979 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Documentation: @stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/siglip2.md [hunk 1]
```diff
@@ -14,225 +14,154 @@ rendered properly in your Markdown viewer.
...
 </div>
 
+# SigLIP2
+
 ## Overview
 
-The SigLIP2 model was proposed in [SigLIP 2: Multilingual Vision-Language Encoders with Improved Semantic Understanding, Localization, and Dense Features](https://huggingface.co/papers/2502.14786) by Michael Tschannen, Alexey Gritsenko, Xiao Wang, Muhammad Ferjad Naeem, Ibrahim Alabdulmohsin,
-Nikhil Parthasarathy, Talfan Evans, Lucas Beyer, Ye Xia, Basil Mustafa, Olivier Hénaff, Jeremiah Harmsen,
-Andreas Steiner and Xiaohua Zhai.
+[SigLIP2](https://huggingface.co/papers/2502.14786) is a family of multilingual vision-language encoders, it is build upon the success of the original [SigLIP](siglip). The original image-text training objective is extended by several independently developed techniques. These include decoder-based pretraining, self-supervised losses such as self-distillation and masked prediction, as well as online data curation.
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:29]:
```suggestion
[SigLIP2](https://huggingface.co/papers/2502.14786) is a family of multilingual vision-language encoders that builds on the [SigLIP](./siglip) training recipe. It includes decoder-based pretraining, self-distillation, and masked prediction to improve dense prediction tasks (segmentation, depth estimation, etc.). This model is available in two variants:

- NaFlex supports different resolutions and maintains the native image aspect ratio
- FixRes supports fixed resolutions and is backwards compatible with [SigLIP](./siglip)
```

File: docs/source/en/model_doc/siglip2.md [hunk 2]
```diff
@@ -14,225 +14,154 @@ rendered properly in your Markdown viewer.
...
 
 The model comes in two variants
+- FixRes - model works with fixed resolution images (backward compatible with SigLIP v1)
+- NaFlex - model works with variable image aspect ratios and resolutions (SigLIP2 in `transformers`)
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:33]:
```suggestion
```

File: docs/source/en/model_doc/siglip2.md [hunk 3]
```diff
@@ -14,225 +14,154 @@ rendered properly in your Markdown viewer.
...
-alt="drawing" width="600"/>
+You can find all the original SigLIP2 checkpoints under the [SigLIP2](https://huggingface.co/collections/google/siglip2-67b5dcef38c175486e240107) collection.
 
-This model was contributed by [qubvel](https://huggingface.co/qubvel-hf).
-The original code can be found [here](https://github.com/google-research/big_vision/tree/main).
+> [!TIP]
+> Click on the SigLIP2 models in the right sidebar for more examples of how to apply SigLIP2 to different image and text tasks.
 
-## Usage example
+The example below demonstrates how to generate similarity scores between texts and image(s) with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:40]:
```suggestion
The example below demonstrates zero-shot classification with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/siglip2.md [hunk 4]
```diff
@@ -14,225 +14,154 @@ rendered properly in your Markdown viewer.
...
->>> image = Image.open(requests.get(url, stream=True).raw)
-
->>> # inference
->>> candidate_labels = ["2 cats", "a plane", "a remote"]
->>> outputs = image_classifier(image, candidate_labels=candidate_labels)
->>> outputs = [{"score": round(output["score"], 4), "label": output["label"] } for output in outputs]
->>> print(outputs)
-[{'score': 0.1499, 'label': '2 cats'}, {'score': 0.0008, 'label': 'a remote'}, {'score': 0.0, 'label': 'a plane'}]
+pipeline = pipeline(task="zero-shot-image-classification", model="google/siglip2-base-patch16-224", device=0, torch_dtype=torch.bfloat16)
+print(pipeline(image, candidate_labels=candidate_labels))
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:53]:
```suggestion
pipeline(image, candidate_labels=candidate_labels)
```

File: docs/source/en/model_doc/siglip2.md [hunk 5]
```diff
@@ -14,225 +14,154 @@ rendered properly in your Markdown viewer.
...
+with torch.no_grad():
+    outputs = model(**inputs)
 
->>> logits_per_image = outputs.logits_per_image
->>> probs = torch.sigmoid(logits_per_image) # these are the probabilities
->>> print(f"{probs[0][0]:.1%} that image 0 is '{candidate_labels[0]}'")
-15.0% that image 0 is '2 cats'
+logits_per_image = outputs.logits_per_image
+probs = torch.sigmoid(logits_per_image)
+print(f"{probs[0][0]:.1%} that image 0 is '{candidate_labels[0]}'")
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:83]: Let's also add the NaFlex example in the `<hfoption id="AutoModel">` block too

```md
<hfoption id="AutoModel">

FixRes example
NaFlex example

</hfoption>
</hfoptions>
```

File: docs/source/en/model_doc/siglip2.md [hunk 6]
```diff
@@ -14,225 +14,154 @@ rendered properly in your Markdown viewer.
...
->>> print(f"{probs[0][0]:.1%} that image 0 is '{candidate_labels[0]}'")
-19.8% that image 0 is '2 cats'
-```
+## Notes
 
+- Training is supported for DDP and FSDP on single-node multi-GPU setups. However, it does not use [torch.distributed](https://pytorch.org/tutorials/beginner/dist_overview.html) utilities which may limit the scalability of batch size.
+- When using the standalone [`GemmaTokenizerFast`] make sure to pass `padding="max_length"` and `max_length=64` as that's how the model was trained.
+- Model was trained with *lowercased* text, make sure you make the same preprocessing for your text labels.
+- To get the same results as the [`Pipeline`], a prompt template of `"This is a photo of {label}."` should be passed to the processor.
+- The NaFlex variant enables processing different types of images at appropriate resolution, e.g. using a larger resolution to process document images, while at the same time minimizing the impact of aspect ratio distortion on certain inference tasks, e.g. on OCR.
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:127]:
```suggestion
- The NaFlex variant processes different types of images at the appropriate resolution (using a larger resolution to process document images for example), while also minimizing the impact of aspect ratio distortion for certain inference tasks like OCR.

   NaFlex resizes the input image so the height and width are multiples of the patch size after resizing. It keeps the aspect ratio distortion as low as possible and produces a sequence length of at most the desired target sequence length (`max_num_patches`). After resizing, the image is split into a sequence of patches and a mask with padding information is added.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for your contribution! :)

File: docs/source/en/model_doc/siglip2.md [hunk 7]
```diff
@@ -14,225 +14,161 @@ rendered properly in your Markdown viewer.
...
-This model was contributed by [qubvel](https://huggingface.co/qubvel-hf).
-The original code can be found [here](https://github.com/google-research/big_vision/tree/main).
+[SigLIP2](https://huggingface.co/papers/2502.14786) is a family of multilingual vision-language encoders that builds on the [SigLIP](./siglip) training recipe. It includes decoder-based pretraining, self-distillation, and masked prediction to improve dense prediction tasks (segmentation, depth estimation, etc.). This model is available in two variants:
 
-## Usage example
+- NaFlex supports different resolutions and maintains the native image aspect ratio
+- FixRes supports fixed resolutions and is backwards compatible with [SigLIP](./siglip)
 
-There are 2 main ways to use SigLIP2: either using the pipeline API, which abstracts away all the complexity for you, or by using the `Siglip2Model` class yourself.
+The model comes in two variants
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:34]:
```suggestion
```

File: docs/source/en/model_doc/siglip2.md [hunk 8]
```diff
@@ -14,225 +14,161 @@ rendered properly in your Markdown viewer.
...
->>> outputs = [{"score": round(output["score"], 4), "label": output["label"] } for output in outputs]
->>> print(outputs)
-[{'score': 0.1499, 'label': '2 cats'}, {'score': 0.0008, 'label': 'a remote'}, {'score': 0.0, 'label': 'a plane'}]
+pipeline = pipeline(task="zero-shot-image-classification", model="google/siglip2-base-patch16-224", device=0, torch_dtype=torch.bfloat16)
+pipeline(image, candidate_labels=candidate_labels)
 ```
 
-**Using the model yourself**
+</hfoption>
+<hfoption id="AutoModel - FixRes">
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:58]: Tiny nit just to make it more compact

```suggestion
<hfoption id="AutoModel (FixRes)">
```

File: docs/source/en/model_doc/siglip2.md [hunk 9]
```diff
@@ -14,225 +14,161 @@ rendered properly in your Markdown viewer.
...
-
->>> candidate_labels = ["2 cats", "2 dogs"]
-# follows the pipeline prompt template to get same results
->>> texts = [f"This is a photo of {label}." for label in candidate_labels]
-
-# default value for `max_num_patches` is 256, but you can increase resulted image resolution providing
-# higher values e.g. `max_num_patches=512`
->>> inputs = processor(text=texts, images=image, max_num_patches=256, return_tensors="pt")
+</hfoption>
+<hfoption id="AutoModel - NaFlex">
```
Comment (stevhliu) [inline docs/source/en/model_doc/siglip2.md:88]:
```suggestion
<hfoption id="AutoModel (NaFlex)">
```

File: docs/source/en/model_doc/siglip2.md (hunk 5 above)
Comment (stevhliu) [reply docs/source/en/model_doc/siglip2.md:84]: Oh thats a really good idea! 👍

**Review by stevhliu — APPROVED** [review]
Awesome, thanks for your work! We can merge after a few more minor nits :)

**Comment by saswatmeher** [comment]
Addressed. Thank you for your suggestions! :)
